lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,64 @@
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "compress package as a zip file", :default => LogStash::Environment.windows?
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+  option "--override", :flag, "Override a previously generated package file", :default => false
+
+  def execute
+    signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+    puts("Packaging plugins for offline usage")
+    if override?
+      rm(target_file)
+    else
+      validate_target_file
+    end
+    LogStash::Bundler.invoke!({:package => true, :all => true})
+    compress_cache(LogStash::Environment::CACHE_PATH)
+    puts("Generated at #{target_file}");
+    rm_rf(LogStash::Environment::CACHE_PATH) if clean?
+  end
+
+  private
+
+  def compress_cache(location)
+    require "logstash/util/compress"
+    if zip?
+      LogStash::Util::Zip.compress(location, target_file)
+    else
+      LogStash::Util::Tar.compress(location, target_file)
+    end
+  end
+
+  def validate_target_file
+    if File.exist?(target_file)
+      puts("File #{target_file} exist, do you want to override it to continue? (Y/N)")
+      override = ( "y" == STDIN.gets.strip.downcase ? true : false)
+      if override
+        File.delete(target_file)
+      else
+        signal_error("Package creation cancelled, a previously generated package exist at location: #{target_file}, move this file to safe place and run the command again")
+        exit
+      end
+    end
+  end
+
+  def target_file
+    target_file = File.join(LogStash::Environment::LOGSTASH_HOME, "plugins_cache")
+    "#{target_file}#{file_ending}"
+  end
+
+  def file_ending
+    zip? ? ".zip" : ".tar.gz"
+  end
+
+  def rm(path)
+    File.delete(path)
+  end
+
+  def rm_rf(dir)
+    require "fileutils"
+    FileUtils.rm_rf(dir)
+  end
```

Comment:
I dont think we need theses method `#rm` and `#rm_rf` ? they don't really provide more value than the actual `FileUtils.rm_rf` or `File.delete`.